src/ring_theory/nullstellensatz.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import ring_theory.jacobson
+import field_theory.algebraic_closure
+import field_theory.mv_polynomial
+
+/-!
+  # Nullstellensatz
+  This file establishes a version of Hilbert's classical Nullstellensatz for `mv_polynomial`s.
+  The main statement of the theorem is `vanishing_ideal_zero_locus_eq_radical`.
+
+  The statement is in terms of new definitions `vanishing_ideal` and `zero_locus`.
+  Mathlib already has versions of these in terms of the prime spectrum of a ring,
+    but those are not well-suited for expressing this result.
+  Suggestions for better ways to state this theorem or organize things are welcome.
+
+  The machinery around `vanishing_ideal` and `zero_locus` is also minimal, I only added lemmas
+    directly needed in this proof, since I'm not sure if they are the right approach.
+-/
+
+open ideal
+open mv_polynomial
+
+noncomputable theory
+
+variables {k : Type*} [field k]
+variables {σ : Type*} [fintype σ]
+
+/-- Set of points that are zeroes of all polynomials in an ideal -/
+def zero_locus (I : ideal (mv_polynomial σ k)) : set (σ → k) :=
```

Comment:
Could you add `lemma mem_zero_locus {I : ideal _} {x} : x ∈ zero_locus I ↔ ∀ p ∈ i, eval x p = 0`? It should probably even be marked `@[simp]`.